core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -334,6 +325,10 @@ private boolean isOverTheLimit() {
         return false;
     }
 
+    private Runnable cancelTask(ThreadPool.Cancellable cancellable) {
+        return cancellable::cancel;
```

Comment:
Yes we can fold it in. And we are not using FutureUtils because we are not cancelling a ScheduledFuture anymore. Repeated tasks scheduled on the ES ThreadPool are implement `ThreadPool.Cancellable`.